platforms/software/maven/src/integTest/groovy/org/gradle/api/publish/maven/MavenPublishExternalVariantIntegrationTest.groovy

```diff
@@ -861,5 +1044,30 @@ class MavenPublishExternalVariantIntegrationTest extends AbstractMavenPublishInt
         output
     }
 
+    /**
+     * Prints the classpath for a compilation with the given name, including the compilation's own artifacts.
+     * {@link #readClasspath(String)} will read the output of this task into a set, which should match the
+     * classpath resolved by Maven.
+     */
+    def printClasspathTask(String name = "") {
+        String runtimeClasspath = name.isEmpty() ? "runtimeClasspath" : "${name}RuntimeClasspath"
+        String runtimeElements = name.isEmpty() ? "runtimeElements" : "${name}RuntimeElements"
+
+        """
+            task print${name.capitalize()}RuntimeClasspath {
+                def artifacts = configurations.${runtimeElements}.outgoing.artifacts.files
+                def classpath = configurations.${runtimeClasspath}.incoming.files
+                doLast {
+                    def allFiles = classpath.files + artifacts.files
+                    println "'${name}' classpath: " + allFiles*.name.join(",")
+                }
+            }
+        """
+    }
+
+    Set<String> readClasspath(String name = "") {
+        output.readLines().find { it.startsWith("'${name}' classpath: ") }.split(":")[1].substring(1).split(",") as Set
+    }
+
```

Comment:
❓ Can we just write this to a file instead?

Reply:
I dont see why not